jupyter-docker/jupyter_delocalize.py

```diff
@@ -0,0 +1,152 @@
+from datetime import datetime, timedelta
+import json
+import os
+import re
+import subprocess
+from notebook.services.contents.filemanager import FileContentsManager
+
+METADATA_TTL = timedelta(minutes=5)
+
+class DelocalizingContentsManager(FileContentsManager):
+  """
+  A contents manager allowing for configurable automated delocalization.
+
+  Files will always be persisted to the local Jupyter file system as usual. The
+  decision of whether to try delocalization is driven by the presence of a file
+  named .delocalize.json on the Jupyter file system. The format of this
```

Comment:
This comment gives me almost all of the details I need, but I had to go back to the doc for one piece—the where. I might reword this to: "... presence of a file named .delocalize.json in the same directory as the notebook file that is the target of the save operation."

Reply:
Good point, thanks. Sent https://github.com/DataBiosphere/leonardo/pull/301

Reworded slightly to capture that it is not only applicable to `save()` operations (also moves, deletes), and also is not restricted to notebooks - could be data files if someone uploaded something and the filename matches the given pattern.